### PR TITLE
Refactor poller implementation

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
To prep for data-plane poller implementation, no functional changes.
Consolidated poller implementations into pollers.ts.
Refactored ARM implementation.
Removed comments on unexported types and methods.
Fixed poller doc comment generation.
Pinned blob storage swagger to prevent inadvertent changes.

### CODE REVIEW

File: test/synapse/2019-06-01/azartifacts/zz_generated_pollers.go [hunk 1]
```diff
@@ -7,74 +7,135 @@
...
 package azartifacts
 
-import "github.com/Azure/azure-sdk-for-go/sdk/azcore"
-
-// CreateDataFlowDebugSessionResponsePoller provides polling facilities until the operation completes
+import (
+	"context"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"net/http"
+	"time"
```
Comment (catalinaperalta) [inline test/synapse/2019-06-01/azartifacts/zz_generated_pollers.go:14]: This isnt used here.

**Comment by catalinaperalta** [comment]
So you think data-plane and arm poller implementations will be more similar than we previously envisioned? I guess my main question is, will there be a big benefit in a shared poller file? Or could it get complicated with more future divisions between ARM and data-plane implementations? Not against it by the way, just want to understand the change.

File: test/synapse/2019-06-01/azartifacts/zz_generated_pollers.go (hunk 1 above)
Comment (jhendrixMSFT) [reply test/synapse/2019-06-01/azartifacts/zz_generated_pollers.go]: Fixed

**Comment by jhendrixMSFT** [comment]
It seems likely.  We'll have a basic `azcore.Poller`, similar to what Python has, that will be used by data-plane.  Both cases need to generate the type-specific poller interfaces and their implementations.  They just have different guts.

**Review by catalinaperalta — APPROVED** [review]
LGTM

**Pull request merged**